### PR TITLE
Cover more cases

### PR DESCRIPTION
**Pull request by chenxsan**
## Why

Currently we only return `node.children[0].value` in https://github.com/montogeek/remark-extract-anchors/blob/master/src/index.js#L30, which means it would miss some data when we have complex structures in headings, for example https://webpack.js.org/api/hot-module-replacement/#accept:

image

And here's the updated version:

image

### CODE REVIEW

File: src/reduceChildren.test.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+const reduce = require('./reduceChildren')
+describe('reduceChildren', () => {
+  it('', () => {
```
Comment (montogeek) [inline src/reduceChildren.test.js]: Please add a name

File: src/reduceChildren.test.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          start: { line: 10, column: 12, offset: 421 },
+          end: { line: 10, column: 21, offset: 430 },
+          indent: []
+        }
+      }
+    ]
+    expect(reduce(data)).toEqual(['Install ', 'webpack'])
+  })
+
+  it('should return all texts within inline code', () => {
```
Comment (montogeek) [inline src/reduceChildren.test.js]: When having `()` it also get included in the value?

Comment (chenxsan) [reply src/reduceChildren.test.js]: `()` alone isn't special to remark, so it's parsed as `text`.

Comment (montogeek) [reply src/reduceChildren.test.js]: Cool

**Comment by montogeek** [comment]
Thanks!

**Comment by montogeek** [comment]
Published on 1.1.1 https://www.npmjs.com/package/remark-extract-anchors/v/1.1.1

**Pull request merged**